src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -380,7 +380,7 @@ public void DiagnosticSourceStartStop()
                     // Confirm that duration is set. 
                     Assert.NotNull(observer.Activity);
                     Assert.True(TimeSpan.Zero < observer.Activity.Duration);
-                    Assert.True(observer.Activity.StartTimeUtc + observer.Activity.Duration < DateTime.UtcNow);
+                    Assert.True(observer.Activity.StartTimeUtc + observer.Activity.Duration <= DateTime.UtcNow.AddTicks(1));
```

Comment:
Right, after #16908 it could be that StartTime is UtcNow and duration is 1 tick, so the asserted condition was false